google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/apis/CloudApiManager.java

```diff
@@ -137,6 +140,28 @@ private static void enableApi(
         .execute();
   }
 
+  /**
+   * Fetches the list of {@link Role} for the supplied {@link CloudProject} by querying the Iam API.
+   */
+  static List<Role> getServiceAccountRoles(CloudProject cloudProject) {
+    Optional<CredentialedUser> user =
+        Services.getLoginService().getLoggedInUser(cloudProject.googleUsername());
+
+    if (!user.isPresent()) {
+      LOG.error("Cannot fetch service account roles: logged in user not found.");
+      return ImmutableList.of();
+    }
+
+    Iam iam = GoogleApiClientFactory.getInstance().getIamClient(user.get().getCredential());
+
+    try {
+      return iam.roles().list().execute().getRoles();
```

Comment:
Could we consider doing this asynchronously as well, or this call usually takes short time and role list is mostly not large?

Reply:
Seems to be very quick. But yes I agree that all of these calls should ultimately be asynchronous. I’ll open an issue to track it separately.